lib/resque/worker.rb

```diff
@@ -484,21 +484,27 @@ def self.all_heartbeats
     def self.all_workers_with_expired_heartbeats
       workers = Worker.all
       heartbeats = Worker.all_heartbeats
+      now = server_time
 
       workers.select do |worker|
         id = worker.to_s
         heartbeat = heartbeats[id]
 
         if heartbeat
-          seconds_since_heartbeat = (Time.now - Time.parse(heartbeat)).to_i
+          seconds_since_heartbeat = (now - Time.parse(heartbeat)).to_i
           seconds_since_heartbeat > Resque.prune_interval
         else
           false
         end
       end
     end
 
-    def heartbeat!(time = Time.now)
+    def self.server_time
+      time, _ = redis.time
```

Comment:
To me this usage seems like it is deprecated based on the comment in DataStore#method_missing (https://github.com/resque/resque/blob/160ac190e94acead7289e9c815d1ef8b0a15843c/lib/resque/data_store.rb#L50-L54).  Also, using `redis` like this seems like it defeats the purpose of https://github.com/resque/resque/commit/4bb44413a045fca78d25e22ad535b3f5ea97fab8.  If we are going to put an abstraction between resque and redis, then this is a leaky abstraction.

method_missing is also slower to use than a normal method call.